dashboard/src/components/Tabs.vue

```diff
@@ -1,15 +1,25 @@
 <script setup lang="ts">
 import type { FunctionalComponent, SVGAttributes } from "vue";
 import type { RouteLocationResolved } from "vue-router/auto";
+import { useRoute } from "vue-router/auto";
+import { isEqual } from "lodash-es";
+
+const route = useRoute();
+
+type Tab = {
+  icon?: FunctionalComponent<SVGAttributes, {}>;
+  text: string;
+  route: RouteLocationResolved;
+  show: boolean;
+};
 
 const { tabs } = defineProps<{
-  tabs: {
-    icon: FunctionalComponent<SVGAttributes, {}>;
-    text: string;
-    route: RouteLocationResolved;
-    show: boolean;
-  }[];
+  tabs: Tab[];
 }>();
+
+function isActive(tab: Tab): boolean {
+  return tab.route.path == route.path && isEqual(tab.route.query, route.query);
```

Comment:
Why is `lodash` needed for this comparison?

Reply:
`tab.route.query` and `route.query` are objects, and when you compare objects in Typescript / Javascript it checks if they are the identical object not that the properties of the object are equivalent, i.e. "(tab.route.query == route.query) => false".

Reply:
https://www.typescriptlang.org/play/?#code/MYewdgzgLgBCBGArAjDAvDA3jaBDKArhAFwwDkAJuAKZkwC+A3AFCiSwKIBM6WOU+IqUo06TZq3AQQAG2oA6GSADmAClVlVnVGgycuASmEwA1DC1IdepIYMxcEfgCcAlmGWmYAIgA88JzAA9AB8XgaMQA